tests/test_dockerfile.py

```diff
@@ -0,0 +1,392 @@
+            "-daemon=1",
+            "-printtoconsole",
+            ]
+    result_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            }
+    hook.test(test_args, test_env, result_args, result_env)
+    assert hook.result == hook.reference, hook.error_msg()
+
+def test_help_debug(hook):
+    """
+    Test option with dash like `-help-debug` if working
+    properly in environment.
+    """
+    test_args = ["dogecoind"]
+    test_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            "DATADIR" : "/dogecoin/.dogecoin",
+            "HELP_DEBUG" : "",
+            }
+
+    result_args = [
+            abs_path("dogecoind"),
+            "-datadir=/dogecoin/.dogecoin",
+            "-help-debug",
+            "-printtoconsole",
+            ]
+    result_env = {
+            "USER" : "dogecoin",
+            "PATH" : os.environ['PATH'],
+            }
+    hook.test(test_args, test_env, result_args, result_env)
+    assert hook.result == hook.reference, hook.error_msg()
+
+def test_files_metadata(host):
+    """Verify all image files generated by the Dockerfile"""
+    dogecoind = host.file(abs_path("dogecoind"))
+    assert dogecoind.user == "dogecoin"
+    assert dogecoind.group == "dogecoin"
+    assert dogecoind.mode == 0o4555
+
+    dogecoinqt = host.file(abs_path("dogecoin-qt"))
+    assert dogecoinqt.user == "dogecoin"
+    assert dogecoinqt.group == "dogecoin"
+    assert dogecoinqt.mode == 0o4555
+
+    dogecointx = host.file(abs_path("dogecoin-tx"))
+    assert dogecointx.user == "dogecoin"
+    assert dogecointx.group == "dogecoin"
+    assert dogecointx.mode == 0o4555
+
+    dogecoincli = host.file(abs_path("dogecoin-cli"))
+    assert dogecoincli.user == "dogecoin"
+    assert dogecoincli.group == "dogecoin"
+    assert dogecoincli.mode == 0o4555
+
+    entrypoint_script = host.file(abs_path("entrypoint.py"))
+    assert entrypoint_script.user == "root"
+    assert entrypoint_script.group == "root"
+    assert entrypoint_script.mode == 0o500
+
+def test_datadir(hook, host):
+    """
+    Verify if datadir can be changed and created proprely.
+
+    Verify datadir metada, if it's owned by script user (dogecoin).
+    """
+    #Use a unique directory for this test
+    test_datadir = "/tmp/datadir_test"
+    datadir_argument = f"-datadir={test_datadir}"
+
+    test_args = ["dogecoind", datadir_argument]
+    test_environ = {
+        "USER" : "dogecoin",
+        "PATH" : os.environ["PATH"],
+            }
+
+    result_args = [
+            abs_path("dogecoind"),
+            datadir_argument,
+            "-printtoconsole",
+            ]
+    result_environ = {
+        "USER" : "dogecoin",
+        "PATH" : os.environ["PATH"],
+            }
+
+    #Run test to generate datadir
+    hook.test(test_args, test_environ, result_args, result_environ)
+    assert hook.result == hook.reference, hook.error_msg()
+
+    #Test datadir metadata
```

Comment:
This must be moved to an integration test too

Reply:
I removed it, keep record of the test in https://github.com/dogecoin/docker/issues/33 to convert it in CI tests.

Reply:
I made a mistake in this comment, I didn't remove this test, it's the previous one. It verifies that datadir is properly created by the entrypoint + chown, it manages the creation of the directory. Don't you think I should keep it here ?